internal/host/validator.go

```diff
@@ -1498,51 +1270,81 @@ func (v *validator) shouldValidateDNSResolution(cluster *common.Cluster) (bool,
 	}
 	if shouldValidate, ret := v.shouldValidateDNSResolution(c.cluster); !shouldValidate {
-		return ret
-	}
-	apiDomainName := domainNameToResolve(c, constants.APIClusterSubdomain)
-	return checkDomainNameResolution(c, apiDomainName)
-}
 
-func (v *validator) printIsAPIDomainNameResolvedCorrectly(c *validationContext, status ValidationStatus) string {
+		if c.cluster != nil {
+			apiDomainName := domainNameToResolve(c, constants.APIClusterSubdomain)
+			return ret, printIsDomainNameResolvedCorrectly(c, ret, apiDomainName, target)
+		}
+		return ret, ""
+	}
 	apiDomainName := domainNameToResolve(c, constants.APIClusterSubdomain)
-	return printIsDomainNameResolvedCorrectly(c, status, apiDomainName, "API")
+	status := checkDomainNameResolution(c, apiDomainName)
+	return status, printIsDomainNameResolvedCorrectly(c, status, apiDomainName, target)
 }
 
-func (v *validator) isAPIInternalDomainNameResolvedCorrectly(c *validationContext) ValidationStatus {
+func (v *validator) isAPIInternalDomainNameResolvedCorrectly(c *validationContext) (ValidationStatus, string) {
+	target := "internal API load balancer"
 	if c.infraEnv != nil {
-		return ValidationSuccessSuppressOutput
+		return ValidationSuccessSuppressOutput, ""
 	}
 	if shouldValidate, ret := v.shouldValidateDNSResolution(c.cluster); !shouldValidate {
-		return ret
+		if c.cluster != nil {
+			apiDomainName := domainNameToResolve(c, constants.InternalAPIClusterSubdomain)
+			return ret, printIsDomainNameResolvedCorrectly(c, ret, apiDomainName, target)
+		}
 	}
 	apiInternalDomainName := domainNameToResolve(c, constants.InternalAPIClusterSubdomain)
-	return checkDomainNameResolution(c, apiInternalDomainName)
+	status := checkDomainNameResolution(c, apiInternalDomainName)
+	return status, printIsDomainNameResolvedCorrectly(c, status, apiInternalDomainName, target)
 }
 
-func (v *validator) printIsAPIInternalDomainNameResolvedCorrectly(c *validationContext, status ValidationStatus) string {
-	apiInternalDomainName := domainNameToResolve(c, constants.InternalAPIClusterSubdomain)
-	return printIsDomainNameResolvedCorrectly(c, status, apiInternalDomainName, "internal API")
-}
-
-func (v *validator) isAppsDomainNameResolvedCorrectly(c *validationContext) ValidationStatus {
+func (v *validator) isAppsDomainNameResolvedCorrectly(c *validationContext) (ValidationStatus, string) {
 	if c.infraEnv != nil {
-		return ValidationSuccessSuppressOutput
+		return ValidationSuccessSuppressOutput, ""
 	}
 	if shouldValidate, ret := v.shouldValidateDNSResolution(c.cluster); !shouldValidate {
-		return ret
+		if c.cluster != nil {
+			return ret, v.printIsAppsDomainNameResolvedCorrectly(c, ret)
+		}
 	}
 	appsDomainName := fmt.Sprintf("%s.apps.%s.%s", constants.AppsSubDomainNameHostDNSValidation, c.cluster.Name, c.cluster.BaseDNSDomain)
-	return checkDomainNameResolution(c, appsDomainName)
+	status := checkDomainNameResolution(c, appsDomainName)
+	return status, v.printIsAppsDomainNameResolvedCorrectly(c, status)
```

Comment:
Same simplification as above could be made here.

Reply:
See above comment

Reply:
It's still relevant. You should be able to deduplicate the `v.printIsAppsDomainNameResolvedCorrectly` call.